fuzzers/baby_fuzzer/src/main.rs

```diff
@@ -26,7 +27,8 @@ fn signals_set(idx: usize) {
 #[allow(clippy::similar_names)]
 pub fn main() {
     // The closure that we want to fuzz
-    let mut harness = |buf: &[u8]| {
+    let mut harness = |input: &BytesInput| {
```

Comment:
Couldn't this be `I: HasBytesVec` then? Or does that get more complex

Reply:
Hm, Ill try and see what the rust compiler says. Maybe it needs a Sized type here. not sure.

Reply:
Hm this would not work as the paramters needs to be an `Input` but input does not extend `HasBytesVec`

Reply:
In that case, why not `Input + HasTargetBytes`?

Reply:
Or, related, why does it need to be an `Input` in this case?

Reply:
BytesInput is fine, the fuzzer use BytesInput

Reply:
I think in this case it is fine to use the concrete type BytesInput here as it is the concrete usage of the LibAFL library.

Reply:
Can't get it work:

```
error[E0225]: only auto traits can be used as additional traits in a trait object
  --> src/main.rs:30:45
   |
30 |     let mut harness = |input: &(dyn Input + HasTargetBytes)| {
   |                                     -----   ^^^^^^^^^^^^^^ additional non-auto trait
   |                                     |
   |                                     first non-auto trait
   |
   = help: consider creating a new trait with all of these as super-traits and using that trait here instead: `trait NewTrait: Input + HasTargetBytes {}`
   = note: auto-traits like `Send` and `Sync` are traits that have special properties; for more information on them, visit <https://doc.rust-lang.org/reference/special-types-and-traits.html#auto-traits>
```